cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -37,7 +37,13 @@ module Raw {
   predicate functionHasIR(Function func) { exists(getTranslatedFunction(func)) }
 
   cached
-  predicate varHasIRFunc(GlobalOrNamespaceVariable var) {
+  predicate varHasIRFunc(Variable var) {
+    (
+      var instanceof GlobalOrNamespaceVariable
+      or
+      not var.isFromUninstantiatedTemplate(_) and
```

Comment:
This makes me wonder: I think we can have global/namespace variables that are uninstantiated templates. Do we already rule out those?

Reply:
Can global variables be templated? 🤔

Reply:
What's possible is things like:
```cpp
template<class T>
T foo = 42;

int main() {
  return foo<int>;
}
```
What I'm asking is whether we exclude the definition of `foo` from IR generation.

Reply:
I totally didn't know that was legal 😂. I think that'll generated bad IR. I'll add a testcase and move the `not var.isFromUninstantiatedTemplate(_)` condition if it turns out to be necessary.

Reply:
I guess the test we want is something like:
```cpp
template<class T>
T foo = 42;

int bar = foo<int>;
```
To ensure we do get relevant code for bar.

Reply:
3d381331e1b44a72d244328d9c7111cefa67d8a0 adds a test. It seems like this Just Works for global variable templates without any exclusion. Maybe because `GlobalOrNamespaceVariable`s (which extend `@globalvariable`) aren't in the `is_variable_template` relation?